scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryUtil.scala

```diff
@@ -36,4 +36,22 @@ object BigQueryUtil {
   /* Generates job ID */
   def generateJobId(projectId: String): String =
     projectId + "-" + UUID.randomUUID().toString
+
+  /** Validates if all selectedFields are in the set of schemaFieldNames */
+  def validateSelectedFields(
+    tableSpec: String,
+    schemaFieldNames: Iterable[String],
+    selectedFields: List[String]
+  ): Unit = {
+    // check if selected fields are defined in the schema of the table
+    if (selectedFields.nonEmpty) {
+      val notFoundFields = selectedFields.toSet -- schemaFieldNames.toSet
```

Comment:
looks good! 

I remembered @sisidra left a comment on #3993 that even though the javadoc doesn't state that nested fields are supported in BQ Storage API, it seems that unofficially, it Just Works ™️    using dot-separated nesting syntax. (https://github.com/spotify/scio/issues/3993#issuecomment-916699743)

I feel a bit hesitant because this doesn't seem to be an official/documented feature but I guess this change would then break Kalvans's code... perhaps it's worth either fully supporting nesting in this `validateSelectedFields` function just in case, or at least validating top-level fields? I'm not sure how nested fields are represented in `bigquery.tables.schema(tableSpec).getFields` though, maybe this is a non issue.

Reply:
ah, right, we need to check how this would work with nested fields. going to run some experiments.